internal/k8s/controllers/http_route_controller_test.go

```diff
@@ -80,6 +83,69 @@ func TestHTTPRoute(t *testing.T) {
 	}
 }
 
+func TestHTTPRouteServiceToRouteRequests(t *testing.T) {
+	t.Parallel()
+
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	svc := &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{Namespace: "namespace-1", Name: "echo-1"},
+	}
+
+	backendNS := gw.Namespace("namespace-1")
+
+	client := gatewayclient.NewTestClient(
+		nil,
+		// Include one route that references the Service name without the namespace,
+		// meaning that the namespace is implicitly the route's namespace
+		&gw.HTTPRoute{
+			ObjectMeta: metav1.ObjectMeta{Namespace: "namespace-1", Name: "route-1"},
+			Spec: gw.HTTPRouteSpec{
+				Rules: []gw.HTTPRouteRule{{
+					BackendRefs: []gw.HTTPBackendRef{
+						{
+							BackendRef: gw.BackendRef{BackendObjectReference: gw.BackendObjectReference{Name: "echo-1"}},
+						},
+						{
+							BackendRef: gw.BackendRef{BackendObjectReference: gw.BackendObjectReference{Name: "echo-2"}},
+						},
+					},
+				}},
+			},
+		},
+		// Include one route in a different namespace that references the Service by
+		// explicit namespace + name
+		&gw.HTTPRoute{
+			ObjectMeta: metav1.ObjectMeta{Namespace: "namespace-2", Name: "route-2"},
+			Spec: gw.HTTPRouteSpec{
+				Rules: []gw.HTTPRouteRule{{
+					BackendRefs: []gw.HTTPBackendRef{{
+						BackendRef: gw.BackendRef{
+							BackendObjectReference: gw.BackendObjectReference{
+								Namespace: &backendNS,
+								Name:      "echo-1",
+							},
+						},
+					}},
+				}},
+			},
+		},
+	)
+
+	controller := &HTTPRouteReconciler{
+		Client:         client,
+		Log:            hclog.NewNullLogger(),
+		ControllerName: mockControllerName,
+		Manager:        reconcilerMocks.NewMockReconcileManager(ctrl),
+	}
+
+	requests := controller.serviceToRouteRequests(svc)
+	require.Len(t, requests, 2)
+	assert.Equal(t, "namespace-1/route-1", requests[0].String())
+	assert.Equal(t, "namespace-2/route-2", requests[1].String())
```

Comment:
I'm not sure if ordering is guaranteed, might be safer to use [`slices.Contains`](https://pkg.go.dev/golang.org/x/exp/slices#Contains) to avoid flakiness.

Reply:
Wound up using `assert.ElementsMatch`